server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -153,14 +229,25 @@ void doRun() {
                 continue;
             }
 
+            boolean isSearchTask = taskCancellation.getTask() instanceof SearchTask;
+
             // Independently remove tokens from both token buckets.
-            boolean rateLimitReached = taskCancellationRateLimiter.get().request() == false;
-            boolean ratioLimitReached = taskCancellationRatioLimiter.get().request() == false;
+            boolean rateLimitReached = isSearchTask
+                ? searchTaskCancellationRateLimiter.get().request() == false
+                : searchShardTaskCancellationRateLimiter.get().request() == false;
+            boolean ratioLimitReached = isSearchTask
+                ? searchTaskCancellationRatioLimiter.get().request() == false
+                : searchShardTaskCancellationRatioLimiter.get().request() == false;
```

Comment:
In continuation to my previous comment.

This can then be simplified to just:
```
TokenBucket rateLimiter = rateLimiters.get(taskCancellation.getTask().getClass());
TokenBucket ratioLimiter = ratioLimiters.get(taskCancellation.getTask().getClass());
```

Reply:
Even if we add limiters to a map, we won't be able to use `taskCancellation.getTask().getClass()` since it fails for the mocked tasks(unittests).

Reply:
Then it means that the mocking isn't done correctly. The mock should return either `SearchTask.class` or `SearchShardTask.class` when `mockTask.getClass()` is called.

Reply:
We are mocking task here [[code](https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/search/backpressure/SearchBackpressureTestHelpers.java#L34)]. 
`mockTask.getClass()` will return something like `org.opensearch.action.search.SearchShardTask$MockitoMock$oQ5hmcox` and not the actual class.

Reply:
As discussed, updated the code and use `instanceOf` to get the value from the map.